### ISSUE TITLE
Terminal minimum contrast ratio not met for dom renderer

### ISSUE THREAD

**Issue by RMacfarlane** [microsoft/vscode#86142]
Issue Type: <b>Bug</b>

Testing https://github.com/microsoft/vscode/issues/85966

1. Use WSL on windows
2. Set `terminal.integrated.minimumContrastRatio` to 21
3. Run true color test from https://unix.stackexchange.com/questions/404414/print-true-color-24-bit-test-pattern/404415#404415

Using the canvas renderer, white and black text are used at this contrast ratio:
![true_color_canvas](https://user-images.githubusercontent.com/3672607/70081095-b7359b00-15bc-11ea-93d8-ad2b4729963e.PNG)

But using the webgl and dom renderer, the true color test is still low contrast:
![true_color_webgl](https://user-images.githubusercontent.com/3672607/70081125-c9173e00-15bc-11ea-904d-e939dc6fc41f.PNG)

VS Code version: Code - Insiders 1.41.0-insider (97855786a014be2440751b038b373c3726e11fe8, 2019-12-03T05:39:36.001Z)
OS version: Windows_NT ia32 10.0.18363

**Comment by Tyriar** [comment]
@RMacfarlane webgl seems to work fine for me, maybe you were still on "dom" because you can't hot swap the webgl one currently (https://github.com/microsoft/vscode/issues/85970)

**Comment by Tyriar** [comment]
https://github.com/xtermjs/xterm.js/issues/2593

**Comment by Tyriar** [comment]
Just cherry picked and manually released xterm.js for this to avoid pulling in some more risky changes

**Comment by RMacfarlane** [comment]
@Tyriar It's now more responsive to the contrast ratio, but it prints entirely white text when I would expect the middle section to be in black:
![true_color_dom](https://user-images.githubusercontent.com/3672607/70268134-af106380-1754-11ea-940e-b999c6e21bf2.PNG)

**Comment by Tyriar** [comment]
☹️ I'll fix this next month as true color isn't used much.

**Comment by Tyriar** [comment]
This is fixed

**Comment by Tyriar** [comment]
![Image](https://user-images.githubusercontent.com/2193314/207090903-d3e57fa6-1641-4651-b3d1-642634457092.png)
